storage/aptosdb/src/pruner/ledger_store/ledger_metadata_pruner.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::schema::{
+    db_metadata::{DbMetadataKey, DbMetadataSchema, DbMetadataValue},
+    version_data::VersionDataSchema,
+};
+use anyhow::{ensure, Result};
+use aptos_schemadb::{ReadOptions, SchemaBatch, DB};
+use aptos_types::transaction::{AtomicVersion, Version};
+use std::sync::{atomic::Ordering, Arc};
+
+#[derive(Debug)]
+pub struct LedgerMetadataPruner {
+    ledger_metadata_db: Arc<DB>,
+    progress: AtomicVersion,
+}
+
+impl LedgerMetadataPruner {
+    pub(in crate::pruner) fn new(ledger_metadata_db: Arc<DB>) -> Result<Self> {
+        let progress = if let Some(v) =
+            ledger_metadata_db.get::<DbMetadataSchema>(&DbMetadataKey::LedgerPrunerProgress)?
+        {
+            v.expect_version()
+        } else {
```

Comment:
I'm not actually quite sure if there is a such case, since we've started writing this progress quite long time ago. Just want to play on a safer side to have this fallback logic.